src/auth/oauth/oauth.ts

```diff
@@ -149,65 +133,33 @@ const ShopifyOAuth = {
       type: DataType.JSON,
       data: body,
     };
+    const cleanShop = sanitizeShop(query.shop, true)!;
 
-    const client = new HttpClient(currentSession.shop);
-    const postResponse = await client.post<OnlineAccessResponse>(postParams);
+    const client = new HttpClient(cleanShop);
+    const postResponse = await client.post(postParams);
```

Comment:
I've removed the casting to `OnlineAccessResponse` (introduced in #447), since the `createSession` method will cast it appropriately anyway (to `OnlineAccessResponse` or `AccessTokenResponse`)